pkg/test/framework/components/istio/ingress.go

```diff
@@ -156,14 +157,17 @@ func (c *ingressImpl) HTTPSAddress() (string, int) {
 }
 
 // DiscoveryAddress returns the externally reachable discovery address (15012) of the component.
-func (c *ingressImpl) DiscoveryAddress() net.TCPAddr {
+func (c *ingressImpl) DiscoveryAddress() netip.AddrPort {
 	host, port := c.AddressForPort(discoveryPort)
-	ip := net.ParseIP(host)
-	if ip.String() == "<nil>" {
+	ip, err := netip.ParseAddr(host)
+	if err != nil {
+		return netip.AddrPort{}
+	}
+	if !ip.IsValid() {
```

Comment:
can ParseAddr really return an invalid IP and err=nil?

Reply:
no, got it, I will remove the `IsValid` check.